src/renderer/components/wallet/txs/send/SendFormBCH.tsx

```diff
@@ -0,0 +1,390 @@
+  balances: WalletBalances
+  balance: WalletBalance
+  onSubmit: (p: SendTxParams) => void
+  isLoading: boolean
+  sendTxStatusMsg: string
+  addressValidation: AddressValidation
+  feesWithRates: FeesWithRatesRD
+  reloadFeesHandler: (memo?: Memo) => void
+  validatePassword$: ValidatePasswordHandler
+  network: Network
+}
+
+export const SendFormBCH: React.FC<Props> = (props): JSX.Element => {
+  const {
+    balances,
+    balance,
+    onSubmit,
+    isLoading,
+    sendTxStatusMsg,
+    addressValidation,
+    feesWithRates: feesWithRatesRD,
+    reloadFeesHandler,
+    validatePassword$,
+    network
+  } = props
+
+  const intl = useIntl()
+
+  const changeAssetHandler = useChangeAssetHandler()
+
+  const [amountToSend, setAmountToSend] = useState<BaseAmount>(ZERO_BASE_AMOUNT)
+
+  const [selectedFeeOptionKey, setSelectedFeeOptionKey] = useState<FeeOptionKey>(DEFAULT_FEE_OPTION_KEY)
+
+  const [form] = Form.useForm<FormValues>()
+
+  const prevFeesWithRatesRef = useRef<O.Option<FeesWithRates>>(O.none)
+
+  const feeRD: FeeRD = useMemo(
+    () =>
+      FP.pipe(
+        feesWithRatesRD,
+        RD.map(({ fees }) => fees[selectedFeeOptionKey])
+      ),
+    [feesWithRatesRD, selectedFeeOptionKey]
+  )
+
+  const oFeesWithRates: O.Option<FeesWithRates> = useMemo(() => FP.pipe(feesWithRatesRD, RD.toOption), [
+    feesWithRatesRD
+  ])
+
+  const feesAvailable = useMemo(() => O.isSome(oFeesWithRates), [oFeesWithRates])
+
+  // Store latest fees as `ref`
+  // needed to display previous fee while reloading
+  useEffect(() => {
+    FP.pipe(
+      oFeesWithRates,
+      O.map((feesWithRates) => (prevFeesWithRatesRef.current = O.some(feesWithRates)))
+    )
+  }, [oFeesWithRates])
+
+  const prevSelectedFeeRef = useRef<O.Option<BaseAmount>>(O.none)
+
+  const selectedFee: O.Option<BaseAmount> = useMemo(
+    () =>
+      FP.pipe(
+        oFeesWithRates,
+        O.map(({ fees }) => {
+          const fee = fees[selectedFeeOptionKey]
+          prevSelectedFeeRef.current = O.some(fee)
+          return fee
+        })
+      ),
+    [oFeesWithRates, selectedFeeOptionKey]
+  )
+
+  const oFeeBaseAmount: O.Option<BaseAmount> = useMemo(
+    () =>
+      FP.pipe(
+        oFeesWithRates,
+        O.map(({ fees }) => fees[selectedFeeOptionKey])
+      ),
+    [oFeesWithRates, selectedFeeOptionKey]
+  )
+
+  const isFeeError = useMemo(() => {
+    return FP.pipe(
+      oFeeBaseAmount,
+      O.fold(
+        // Missing (or loading) fees does not mean we can't sent something. No error then.
+        () => false,
+        (fee) => balance.amount.amount().isLessThan(fee.amount())
+      )
+    )
+  }, [balance.amount, oFeeBaseAmount])
+
+  const renderFeeError = useMemo(() => {
+    if (!isFeeError) return <></>
+
+    const msg = intl.formatMessage(
+      { id: 'wallet.errors.fee.notCovered' },
+      {
+        balance: formatAssetAmountCurrency({
+          amount: baseToAsset(balance.amount),
+          asset: AssetBTC,
```

Comment:
is it correct to use BTC asset here ?

Reply:
Oh no, it should be `AssetBCH` of course - another good catch!